src/ocean/util/app/ext/TaskExt.d

```diff
@@ -0,0 +1,203 @@
+                configuration to init it
+
+    ***************************************************************************/
+
+    public void enable ( typeof(SchedulerConfiguration.specialized_pools)
+            optional_config = null )
+    {
+        this.optional_config = optional_config;
+        this.enabled_ = true;
+    }
+
+    /***************************************************************************
+
+        Parse the configuration file options to get the scheduler configuration
+
+        Params:
+            app = the application instance
+            parser = configuration instance
+
+    ***************************************************************************/
+
+    public override void processConfig ( IApplication app, ConfigParser parser )
+    {
+        if (!this.enabled)
+            return;
+
+        static struct PartialSchedulerConfiguration
+        {
+            size_t worker_fiber_stack_size =
+                SchedulerConfiguration.worker_fiber_stack_size.init;
+            size_t worker_fiber_limit =
+                SchedulerConfiguration.worker_fiber_limit.init;
+            size_t task_queue_limit =
+                SchedulerConfiguration.task_queue_limit.init;
+            size_t suspended_task_limit =
+                SchedulerConfiguration.suspended_task_limit.init;
+        }
+
+        // can't fill SchedulerConfiguration directly because it contains
+        // `PoolDescription` array which has to be provided via code
+
+        SchedulerConfiguration scheduler_config;
+        PartialSchedulerConfiguration partial_config;
+        // fill with init values
+        foreach (idx, ref elem; partial_config.tupleof)
+        {
+            mixin("elem = scheduler_config." ~
+                fieldIdentifier!(typeof(partial_config), idx) ~ ";");
+        }
+
+        // fill with whatever is in config file
+        fill!(PartialSchedulerConfiguration)(
+            "SCHEDULER", partial_config, parser);
+
+        // write back
+        foreach (idx, ref elem; partial_config.tupleof)
+        {
+            mixin("scheduler_config." ~
+                fieldIdentifier!(typeof(partial_config), idx) ~
+                " = elem;");
+        }
+        scheduler_config.specialized_pools = this.optional_config;
+
+        initScheduler(scheduler_config);
+    }
+
+    /***************************************************************************
+
+        Wraps given delegate returning app exit status code in a new allocated
+        task object and immediately starts the scheduler event loop to handle
+        it.
+
+        Params:
+            dg = delegate forwarding to the app entry point
+
+    ***************************************************************************/
+
+    public int run ( int delegate () dg )
+    {
+        int result = -1;
+
+        auto task = new class Task {
+            override void run ( )
+            {
+                result = dg();
```

Comment:
What would it happen if the `dg()` throws?

Reply:
Same as with any other throwing task - depends on the state of the https://github.com/sociomantic-tsunami/ocean/blob/3282758547765ac4dcceca1c81a42a49e8f138e5/src/ocean/task/Scheduler.d#L164-L175

By default it simply rethrows the exception and that will propagate to whatever context has resumed the task. I have updated the test to show one such case.